x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -343,7 +344,7 @@ public void authorizeIndexAction(
                     // the same as the user that submitted the original request so no additional checks are needed here.
                     listener.onResponse(IndexAuthorizationResult.ALLOW_NO_INDICES);
                 }
-            } else if (action.equals(ClosePointInTimeAction.NAME)) {
+            } else if (action.equals(ClosePointInTimeAction.NAME) || action.equals(TransportUnpromotableShardRefreshAction.NAME)) {
```

Comment:
I don't think we should add another security bypass like this. The user should be authorised to refresh the indices to which they're writing, and if they aren't then we should respect that.

Reply:
This is resolved with the new commit I will push. There's no such exception anymore.

Reply:
Still seems to be the case - let's keep this unresolved until the fix is pushed.

Reply:
@DaveCTurner done!